CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+    else ()
+        set(DUCC_ARCH_FLAGS "-march=native" CACHE STRING "Compiler flags for specifying target architecture.")
+    endif ()
+    message(STATUS "Using GCC/Clang flags: ${DUCC_ARCH_FLAGS}")
+else ()
+    # Check for AVX, AVX512 and SSE support
+    message(STATUS "Checking for AVX, AVX512 and SSE support")
+    try_run(RUN_RESULT_VAR COMPILE_RESULT_VAR
+            ${CMAKE_BINARY_DIR}
+            ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CheckAVX.cpp
+            COMPILE_OUTPUT_VARIABLE COMPILE_OUTPUT
+            RUN_OUTPUT_VARIABLE RUN_OUTPUT)
+
+    if (RUN_OUTPUT MATCHES "AVX512")
+        set(DUCC_ARCH_FLAGS "/arch:AVX512" CACHE STRING "Compiler flags for specifying target architecture.")
+    elseif (RUN_OUTPUT MATCHES "AVX")
+        set(DUCC_ARCH_FLAGS "/arch:AVX" CACHE STRING "Compiler flags for specifying target architecture.")
+    elseif (RUN_OUTPUT MATCHES "SSE")
+        set(DUCC_ARCH_FLAGS "/arch:SSE" CACHE STRING "Compiler flags for specifying target architecture.")
+    else ()
+        set(DUCC_ARCH_FLAGS "" CACHE STRING "Compiler flags for specifying target architecture.")
+    endif ()
+    message(STATUS "CPU supports: ${RUN_OUTPUT}")
+    message(STATUS "Using MSVC flags: ${DUCC_ARCH_FLAGS}")
+endif ()
+
+# It is possible to add a sphinx documentation here
+option(DUCC_USE_THREADS "Use threads for parallelization" ON)
+
+add_library(ducc0 STATIC
+        src/ducc0/healpix/healpix_base.cc
+        src/ducc0/healpix/healpix_tables.cc
+        src/ducc0/math/gl_integrator.cc
+        src/ducc0/math/pointing.cc
+        src/ducc0/math/gridding_kernel.cc
+        src/ducc0/math/geom_utils.cc
+        src/ducc0/math/wigner3j.cc
+        src/ducc0/math/space_filling.cc
+        src/ducc0/wgridder/wgridder.cc
+        src/ducc0/infra/string_utils.cc
+        src/ducc0/infra/communication.cc
+        src/ducc0/infra/types.cc
+        src/ducc0/infra/system.cc
+        src/ducc0/infra/threading.cc
+        src/ducc0/infra/mav.cc
+        # add more source files here
+)
+
+target_compile_features(ducc0 PRIVATE cxx_std_17)
+target_include_directories(ducc0 PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>)
+target_include_directories(ducc0 SYSTEM INTERFACE $<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/src>)
+target_compile_options(ducc0 PRIVATE
+        -ffp-contract=fast
+        -fexcess-precision=fast
+        -fno-math-errno
+        -fno-signed-zeros
+        -fno-trapping-math
+        -fassociative-math
+        -freciprocal-math
+        # if nan are not used we could disable them manually
+        # Why no -fast-math or -funsafe-math-optimizations ?
+        # It breaks the code, and influences other software that depend on ducc by changing the rounding mode.
+        # GCC-13 recently fixed this issue:
+        # https://github.com/llvm/llvm-project/issues/57589
+        # https://gcc.gnu.org/gcc-13/changes.html
+        # https://trofi.github.io/posts/302-Ofast-and-ffast-math-non-local-effects.html
```

Comment:
To be fully precise, *compiling* with `-ffast-math` is OK, even with older compilers; *linking* is the problem.
Still, I tried to replace `-ffast-math` with all its sub-flags, just to be more explicit about the optimizations I can live with, but unfortunately this is not portable, since some non-gcc compilers don't understand some of the flags (I think one example was `clang` and `-fcx-limited-range`).
So I'm still using `-ffast-math` in `ducc`'s `setup.py`, but I'm happy to work with the individul flags here and refine the set for other platforms if necessary.